libindy/src/services/pool/events.rs

```diff
@@ -199,8 +201,13 @@ impl Into<Option<RequestEvent>> for PoolEvent {
                 _parse_msg(&msg).map(|parsed|
                     match parsed {
                         //TODO change mapping for CatchupReq. May be return None
-                        //TODO: REMOVE UNWRAP!!!!!
-                        Message::CatchupReq(_) => RequestEvent::CatchupReq(MerkleTree::from_vec(Vec::new()).unwrap(), 0, vec![]),
+                        Message::CatchupReq(_) => RequestEvent::CatchupReq(
+                            MerkleTree {
```

Comment:
Maybe here would be better to use a `Default` trait and implement it for Merkle tree?

Reply:
There is already a Default impl for MerkleTree in request_handler.rs which is different to that would be needed here. It is used 20 times in the code. It has the empty vector as the hash while this impl has the hash of the empty vector as the hash.